src/components/Nav.js

```diff
@@ -1,28 +1,54 @@
-import React from 'react';
-
+import { useContext } from 'react';
 import { NavLink } from 'react-router-dom';
+import { MapCenterContext } from '../context/MapCenterContext';
+import FlyToMeButton from './FlyToMeButton';
+import ShareMyLocationButton from './ShareMyLocationButton';
+
+const Nav = () => {
+  const valueCenterMap = useContext(MapCenterContext);
+  const { trackingId, userLocationShared } = valueCenterMap;
+  const wasWatchingLocation = trackingId !== null;
 
-const Nav = () => (
-  <footer>
-    <nav>
-      <ul className="nav">
-        <li>
-          <NavLink exact to="/">
-            Map
-          </NavLink>
-        </li>
-        <li>
-          <NavLink to="/list">List</NavLink>
-        </li>
-        <li>
-          <NavLink to="/settings">Settings</NavLink>
-        </li>
-        <li>
-          <NavLink to="/help">Help</NavLink>
-        </li>
-      </ul>
-    </nav>
-  </footer>
-);
+  return (
+    <footer>
+      <nav>
+        <ul className="nav">
+          <li>
+            <NavLink exact to="/">
+              Map
+            </NavLink>
+          </li>
+          <li>
+            <NavLink to="/list">List</NavLink>
+          </li>
+          <li>
+            <NavLink to="/settings">Settings</NavLink>
+          </li>
+          <li>
+            <NavLink to="/help">Help</NavLink>
+          </li>
+          {!wasWatchingLocation ? (
+            <li>
+              <ShareMyLocationButton />
+            </li>
+          ) : (
+            <li>
+              <ShareMyLocationButton disabled={true} />
+            </li>
+          )}
+          {userLocationShared ? (
+            <li>
+              <FlyToMeButton />
+            </li>
+          ) : (
+            <li>
+              <FlyToMeButton disabled={true} />
```

Comment:
This works well and it seems like after today's meeting we could change this buttons to be hidden rather than disabled if not in use but we can add a new task to do this later if needed.

Reply:
Agreed. Possibly changing how the `Share Location` and `Fly To Me` buttons are shown (or not shown) when not useable is a great suggestion for a separate task.

Reply:
I will write up a ticket for this 🤓